src/l1block.rs

```diff
@@ -248,23 +248,123 @@ impl L1BlockInfo {
             .wrapping_div(TX_L1_FEE_PRECISION_U256) // account for penalty
     }
 
+    fn calculate_tx_l1_cost_galileo(
+        &self,
+        tx_size: u32, // size of the original rlp-encoded transaction
+        spec_id: ScrollSpecId,
+        compressed_size: u32, // size of the compressed rlp-encoded transaction
+    ) -> U256 {
+        // Post Galileo rollup fee formula:
+        // rollup_fee(tx) = fee_per_byte * compressed_size(tx) * (1 + penalty(tx)) / PRECISION
+        //
+        // Where:
+        // fee_per_byte = (exec_scalar * l1_base_fee + blob_scalar * l1_blob_base_fee)
+        // compressed_size(tx) = min(len(zstd(rlp(tx))), len(rlp(tx)))
+        // penalty(tx) = compressed_size(tx) / penalty_factor
+
+        assert!(
+            compressed_size <= tx_size,
+            "transaction compressed size {compressed_size} must be less than or equal to the original size {tx_size}"
+        );
+
+        let compressed_size = U256::from(compressed_size);
+
+        let exec_scalar = self
+            .l1_commit_scalar
+            .unwrap_or_else(|| panic!("missing exec scalar in spec_id={spec_id:?}"));
+
+        let blob_scalar = self
+            .l1_blob_scalar
+            .unwrap_or_else(|| panic!("missing l1 blob scalar in spec_id={spec_id:?}"));
+
+        let l1_blob_base_fee = self
+            .l1_blob_base_fee
+            .unwrap_or_else(|| panic!("missing l1 blob base fee in spec_id={spec_id:?}"));
+
+        let penalty_factor = match self.penalty_factor {
+            Some(f) if f == U256::from(0) => U256::from(1), // sanitize zero penalty factor
```

Comment:
Why need this transfer? how about also like `l1_blob_base_fee`, just panic if we don't have the value